content/blog/custom-docker-images-for-emr-serverless.md

```diff
@@ -0,0 +1,199 @@
+
+Our second use case is the installation of all custom libraries (in the form of jars etc.,) used by the code. We provide step by step walkthrough of how we build custom images addressing both the above scenarios, use the ECR to register our docker image, and finally use the docker image in the EMR Serverless.
+
+*Step 1:*  Pre-requisites
+
+*Step 2:*  Identify the base EMR Serverless Docker Image, packages/jars that need to be installed/downloaded, and prepare the custom docker Image.
+
+*Step 3:*  Push the Docker Image to the ECR
+
+*Step 4:* Create the EMR Serverless applications via
+
+(a) Command using AWS CLI
+
+(b) through settings in the UI when creating or updating the EMR serverless application.
+
+## Step 1: Prerequisites to build a custom docker image:
+
+* The latest version of the docker client
+* The latest version of AWS CLI
+* Verify the access to dependent repository from where resources (modules, code etc) might be downloaded.   
+
+## **Step 2: Sample Docker file used to create custom EMRServerless image**
+
+<!--StartFragment-->
+
+*\# Refer AWS EMR documentation for the release version tag*\
+*FROM public.ecr.aws/emr-serverless/spark/emr-6.9.0:20221108*
+
+*USER root*
+
+*\# Dependent JAR files*\
+*RUN curl -O <https://repo1.maven.org/maven2/io/delta/delta-core_2.12/2.2.0/delta-core_2.12-2.2.0.jar>*\
+*RUN curl -O <https://repo1.maven.org/maven2/io/delta/delta-storage/2.2.0/delta-storage-2.2.0.jar>*
+
+*\# The base emr-image sets WORKDIR to /home/hadoop, hence the JAR files will be downloaded under /home/hadoop.*\
+*\# Then these jars will be copied to /usr/lib/spark/jars which was set as SPARK_HOME by EMR base image.*\
+*RUN cp /home/hadoop/delta-core_2.12–2.2.0.jar /usr/lib/spark/jars/*\
+*RUN cp /home/hadoop/delta-storage-2.2.0.jar /usr/lib/spark/jars/*
+
+*\# EMRS will run the image as hadoop*\
+*USER hadoop:hadoop*
+
+<!--EndFragment-->
+
+## **Step 3: Pushing the docker image to the AWS ECR:**
+
+Command to authenticate Docker to an AWS ECR public registry assuming region as us-east-1. As EMRServerless base docker image is present in ECR public registry, run below command before building your customized docker file.
+
+<!--StartFragment-->
+
+***aws ecr-public get-login-password — region us-east-1 — profile <profile_name_in_your_aws_credentials_file> | docker login — username AWS — password-stdin public.ecr.aws***
```

Comment:
For commands syntax throughout the article, it would be better to use code block snippet with appropriate language syntax rather than blog-italic syntax through the blog post. Language syntax could be "shell" , "bash", or "markdown" for example.